src/vanilla.ts

```diff
@@ -125,9 +125,13 @@ const buildProxyFunction = (
         return
       }
       const value = Reflect.get(target, key)
+      const { enumerable = false } = Reflect.getOwnPropertyDescriptor(
```

Comment:
If `undefined` is acceptable by TypeScript, let's go with it. (We just avoid using the knowledge what is the default.)
```suggestion
      const { enumerable } = Reflect.getOwnPropertyDescriptor(
```

Reply:
Maybe not? https://github.com/pmndrs/valtio/pull/800#discussion_r1366825175 seems to  say so.

Reply:
````javascript
const { enumerable } = Reflect.getOwnPropertyDescriptor(
        target,
        key
      ) as PropertyDescriptor
      const desc: PropertyDescriptor = {
        value,
        enumerable: enumerable as boolean,
        // This is intentional to avoid copying with proxy-compare.
        // It's still non-writable, so it avoids assigning a value.
        configurable: true,
      }
````
As I understand, this is correct from JavaScript view and doesn't rely on `defineProperty` defaults.
`as boolean` silences TypeScript (although, is not strictly correct type-wise)